User authentication login endpoint added

agusflores opened the pull request:
Feature/US-32

Cree un endpoint de autenticacion de login del usuario en el controller.
Cree un metodo en la interfaz del business para identificar usuario por email y otro metodo para validar al usuario antes de logear.
Desarrolle ambos metodos en el business. 
Cree una clase UserLoginDTO dentro de la carpeta UserDTO.

Adjunto imagenes

Caso 1: Ingreso email de un usuario que no esta registrado 

![image](https://user-images.githubusercontent.com/88952763/186970781-1daa5b31-dde3-4c8d-a776-f72d17a5f274.png)

Caso 2:  Ingreso un email erroneo (agregando dos "@" o sacando el ".com")
![image](https://user-images.githubusercontent.com/88952763/186970831-47517faa-dc5c-4117-b5d7-900b8acff547.png)

Caso 3: Deberia logearse con un usuario registrado pero el endpoint esta mal creado ya que no registra correctamente a los users.

agusflores committed:
User authentication login endpoint added

Feature/US-32

agusflores committed:
User password maxlenght changed 

Cambio caracteres maximos en la password ya que al guardarse encriptada seran mas caracteres.

emmanuelranone reviewed (changes requested):
Cuando se resuelva la historia que te está frenando, enviame de nuevo los cambios con las capturas que evidencien el funcionamiento

agusflores committed:
Trayendo ultima version del dev a la branch

agusflores committed:
Some changes and fixes 

En el initialdata seeding se ingreso dos veces el seed de Role y en el context se cargo mal una tabla.

agusflores commented:
Pruebas de que funciona el login: 

Caso 1 (se ingresa correctamente mail y password) por lo que retorna al usuario: 

![image](https://user-images.githubusercontent.com/88952763/187057566-16ea9c5f-7047-4930-8174-0a4576c65944.png)

Caso 2 (se ingresa bien el mail pero no la password) por lo que retorna Ok(false) como era solicitado: 

![image](https://user-images.githubusercontent.com/88952763/187057574-031615eb-5221-4a09-8808-e1102d2e3b73.png)

agusflores committed:
Trayendo ultima version del dev a la branch

agusflores committed:
Trayendo la ultima version del dev

agusflores committed:
User login endpoint finished 

Retorna el token generado para ese usuario como se solicitaba. 
Hice algunos arreglos que no compilaban en el business

agusflores committed:
AuthBusiness added

agusflores committed:
Incluyendo role para que funcione correctamente el token

agusflores commented:
Adjunto imagen del token

![image](https://user-images.githubusercontent.com/88952763/187322169-cb96ff01-0f48-43ad-8fe6-0cd0874de910.png)
